src/providers/ldap/sdap.c

```diff
@@ -503,131 +529,105 @@ int sdap_parse_entry(TALLOC_CTX *memctx,
-                          "Missing value after ldap_get_values() ??\n");
-                    ldap_value_free_len(vals);
-                    ret = EINVAL;
+            DEBUG(SSSDBG_TRACE_LIBS, "%s[%d]: '%s'\n",
+                  attr_name, i, setof_attr_val[i].bv_val);
+
+            if (base64) {
+                v.data = sss_base64_enc_ber(attrs, setof_attr_val[i]);
+                if (!v.data) {
+                    ret = ENOMEM;
                     goto done;
                 }
-                for (i = 0; vals[i]; i++) {
-                    if (vals[i]->bv_len == 0) {
-                        DEBUG(SSSDBG_TRACE_LIBS,
-                              "Value of attribute [%s] is empty. "
-                               "Skipping this value.\n", str);
-                        continue;
-                    }
-                    if (base64) {
-                        v.data = (uint8_t *) sss_base64_encode(attrs,
-                                 (uint8_t *) vals[i]->bv_val, vals[i]->bv_len);
-                        if (!v.data) {
-                            ldap_value_free_len(vals);
-                            ret = ENOMEM;
-                            goto done;
-                        }
-                        v.length = strlen((const char *)v.data);
-                    } else {
-                        v.data = (uint8_t *)vals[i]->bv_val;
-                        v.length = vals[i]->bv_len;
-                    }
-                    PROBE(SDAP_PARSE_ENTRY, str, v.data, v.length);
-
-                    if (map) {
-                        /* The same LDAP attr might be used for more sysdb
-                         * attrs in case there is a map. Find all that match
-                         * and copy the value
-                         */
-                        for (ai = base_attr_idx; ai < attrs_num; ai++) {
-                            /* check if this attr is valid with the chosen
-                             * schema */
-                            if (!map[ai].name) continue;
-
-                            /* check if it is an attr we are interested in */
-                            if (strcasecmp(base_attr, map[ai].name) == 0) {
-                                ret = sysdb_attrs_add_val(attrs,
-                                                          map[ai].sys_name,
-                                                          &v);
-                                if (ret) {
-                                    ldap_value_free_len(vals);
-                                    goto done;
-                                }
-                            }
-                        }
-                    } else {
-                        /* No map, just store the attribute */
-                        ret = sysdb_attrs_add_val(attrs, name, &v);
-                        if (ret) {
-                            ldap_value_free_len(vals);
+                v.length = strlen((const char *)v.data);
```

Comment:
Just a reminder of the use of an Octet String (possibly a non 0-terminated string) being used as a C-string.